cmake/modules/AddSwift.cmake

```diff
@@ -356,10 +393,11 @@ function(_add_variant_link_flags)
   endif()
 
   if(NOT "${SWIFT_${LFLAGS_SDK}_ICU_UC}" STREQUAL "")
-    list(APPEND result "-L${SWIFT_${sdk}_ICU_UC}")
+    link_directories("${SWIFT_${sdk}_ICU_UC}")
```

Comment:
I actually think these might just be dead code, as we parse an absolute path to target_link_libraries or whatever it is, so these flags may be unecessary, but that's for another cleanup PR